lib/src/components/NxTabs/__tests__/NxTabs.test.tsx

```diff
@@ -111,38 +84,47 @@ describe('NxTabs', function () {
       </NxTabs>
     );
 
-    expect(component.find('[role="tab"]').first()).not.toHaveClassName('active');
-    expect(component.find('[role="tab"]').last()).toHaveClassName('active');
-    expect(component.find('[role="tabpanel"]')).toHaveText('Content 1');
+    const tabs = getAllByRole('tab');
+    expect(tabs[1]).toHaveTextContent('Tab 1');
+    expect(tabs[0].classList.contains('active')).toBe(false);
+    expect(tabs[1].classList.contains('active')).toBe(true);
+    expect(getByRole('tabpanel')).toHaveTextContent('Content 1');
   });
 
-  it('calls onTabSelect when a tab is clicked', function () {
+  it('calls onTabSelect when a tab is clicked', async function() {
+    const user = userEvent.setup();
     const onTabSelect = jest.fn();
-    const component = mount(
-      <NxTabs activeTab={0} onTabSelect={onTabSelect}>
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-      </NxTabs>
-    );
 
-    component.find('[role="tab"]').simulate('click');
+    const { getByRole } = quickRender({
+      activeTab: 0,
+      onTabSelect
+    });
+
+    await user.click(getByRole('tab'));
 
     expect(onTabSelect).toHaveBeenCalledWith(0);
   });
 
-  it('calls onTabSelect when a tab is selected via keyboard', function () {
+  it('calls onTabSelect when a tab is selected via keyboard', async function() {
+    const user = userEvent.setup();
     const onTabSelect = jest.fn();
-    const component = mount(
-      <NxTabs activeTab={0} onTabSelect={onTabSelect}>
+
+    const { getAllByRole } = quickRender({
+      activeTab: 0,
+      onTabSelect,
+      children: (
         <NxTabList>
           <NxTab>Tab 0</NxTab>
+          <NxTab>Tab 1</NxTab>
         </NxTabList>
-      </NxTabs>
-    );
+      )
+    });
 
-    component.find('[role="tab"]').simulate('keypress', { key: ' ' });
+    const firstTab = getAllByRole('tab')[0];
 
-    expect(onTabSelect).toHaveBeenCalledWith(0);
+    firstTab.focus();
+    await user.keyboard('{ArrowRight}{Enter}');
+
+    expect(onTabSelect).toHaveBeenCalledWith(1);
   });
 });
```

Comment:
Needs tests of the `aria-controls` relationship between tabs and tabpanels

Reply:
> Needs tests of the `aria-controls` and `aria-labelledby` relationships between tabs and tabpanels

Added here: 34ce2797159b5caf66689b8c31a31c1a92ef9a3f